apps/src/applab/manageAssets.jsx

```diff
@@ -0,0 +1,138 @@
+var React = require('react');
+var AssetsApi = require('./manageAssets/clientApi');
+var AssetRow = require('./manageAssets/assetRow.jsx');
+
+var errorMessages = {
+  415: 'This type of file is not supported.',
+  500: 'The server responded with an error.'
+};
+
+function getErrorMessage(status) {
+  return errorMessages[status] || 'An unknown error occurred.';
+}
+
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func,
+    typeFilter: React.PropTypes.string
+  },
+
+  getInitialState: function () {
+    return {
+      assets: null,
+      uploadStatus: ''
+    };
+  },
+
+  componentWillMount: function () {
+    // TODO: Use Dave's client api when it's finished.
+    AssetsApi.ajax('GET', '', function (xhr) {
+      var assets = JSON.parse(xhr.responseText);
+      if (this.props.typeFilter) {
+        assets = assets.filter(function (asset) {
+          return asset.category === this.props.typeFilter;
+        }.bind(this));
+      }
+      this.setState({assets: assets});
+    }.bind(this), function (xhr) {
+      this.setState({uploadStatus: 'Error loading asset list: ' +
+          getErrorMessage(xhr.status)});
+    }.bind(this));
+  },
+
+  fileUploadClicked: function () {
+    var uploader = document.querySelector('#uploader');
```

Comment:
FYI, just noticed the following from https://facebook.github.io/react/docs/component-api.html#getdomnode 

getDOMNode is deprecated and has been replaced with React.findDOMNode().
This method is not available on ES6 class components that extend React.Component. It may be removed entirely in a future version of React.

Makes me wonder if depending on getDOMNode is not the right approach.